qtt/algorithms/fitting.py

```diff
@@ -34,11 +36,33 @@ def _estimate_fermi_model_center(x_data, y_data_linerized):
     else:
         estimated_center_xdata = x_data[estimated_index]
 
-    step_size = (np.mean(y_filtered[center_index:]) - np.mean(y_filtered[:center_index]))
-    amplitude_step = -1.0 * np.sign(y_filtered[estimated_index]) * step_size
+    split_offset = int(np.floor(x_data.size/10))
+    mean_right = np.mean(y_data_linearized[(center_index+split_offset):])
+    mean_left = np.mean(y_data_linearized[:(center_index-split_offset)])
+    amplitude_step = -(mean_right - mean_left)
+    
+    if np.sign(-y_derivative_filtered[estimated_index]) != np.sign(amplitude_step):
+        warnings.warn('step size might be incorrect')
+        
+    if fig is not None:
```

Comment:
I think it would be more readable if you move the plotting to a separate function.